Scripts/Source/UD_AbadonQuest_script.psc

```diff
@@ -115,15 +122,93 @@ String[] Property UD_AbadonSuitNames
+    endif
+    Float loc_arousal = UDmain.UDOM.getArousal(akTarget)
+    if !akTarget.wornhaskeyword(libs.zad_deviousheavybondage)
+        UDCDmain.DisableActor(akTarget)
+        if RandomInt(1,99) < Round(UDCDMain.UD_BlackGooRareDeviceChance*fRange(loc_arousal/50.0,1.0,2.0))
+            ;rare devices, drop more loot and goo
+            if !akTarget.wornhaskeyword(libs.zad_deviousSuit)
+                int random = RandomInt(1,3)
+                if random == 1
+                    libs.LockDevice(akTarget,UDlibs.AbadonBlueArmbinder)
+                elseif random == 2
+                    libs.LockDevice(akTarget,UDlibs.MageBinder)
+                elseif random == 3
+                    libs.LockDevice(akTarget,UDlibs.RogueBinder)
+                endif
+            else
+                int random = RandomInt(1,2)
+                if random == 1
+                    libs.LockDevice(akTarget,UDlibs.AbadonBlueArmbinder)
+                elseif random == 2
+                    libs.LockDevice(akTarget,UDlibs.RogueBinder)
+                endif
+            endif
+            if IsPlayer(akTarget)
+                UDmain.ShowMessageBox("Black goo covers your body and tie your hands while changing shape to RARE bondage restraint!")
+            endif
+        else
+            int random = _NextHandRestrain
+            if !akTarget.wornhaskeyword(libs.zad_deviousSuit)
+                if random == 0
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakArmbinder)
+                elseif random == 1
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakElbowbinder)
+                elseif random == 2
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakYoke)
+                elseif random == 3
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakStraitjacket)
+                elseif random == 4
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakBoxbinder)
+                endif
+                
+            else
+                if random == 0 || random == 3
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakArmbinder)
+                elseif random == 1 
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakElbowbinder)
+                elseif random == 2 || random == 4
+                    libs.LockDevice(akTarget,UDlibs.AbadonWeakYoke)
+                endif
+            endif
+            _NextHandRestrain += 1
+            if _NextHandRestrain > 4
+                _NextHandRestrain = 0 ;loop
+            endif
+            if IsPlayer(akTarget)
+                UDmain.ShowMessageBox("Black goo covers your body and tie your hands while changing shape to bondage restraint!")
+            endif
+        endif
+        
+        ;chance to lock plugs and belt. Only works if player first finished abadon quest (as some plugs can evolve in to abadon plug)
+        if IsCompleted() && RandomInt(1,99) < Round(15.0*fRange(loc_arousal/25.0,1.0,3.5))
```

Comment:
As stated in previous comment, this quest completion check can be moved into the lock plugs function. Thus even without quest it still can allow to get other plugs.